inference-engine/src/mkldnn_plugin/nodes/ctc_loss.cpp

```diff
@@ -0,0 +1,244 @@
+                break;
+            }
+            default: {
+                if (resp) {
+                    std::string errorMsg = "CTCLoss layer does not support precision '"
+                            + std::string(inputs[1]->getTensorDesc().getPrecision().name()) + "'";
+                    errorMsg.copy(resp->msg, sizeof(resp->msg) - 1);
+                }
+            }
+        }
+
+        return OK;
+    }
+
+protected:
+    template<typename I1>
+    StatusCode processData(
+                std::vector<Blob::Ptr>& inputs,
+                std::vector<Blob::Ptr>& outputs,
+                ResponseDesc* resp) noexcept {
+        switch (inputs[2]->getTensorDesc().getPrecision()) {
+            case Precision::I32: {
+                return processData<I1, PrecisionTrait<Precision::I32>::value_type>(inputs, outputs, resp);
+            }
+            case Precision::I64: {
+                return processData<I1, PrecisionTrait<Precision::I64>::value_type>(inputs, outputs, resp);
+            }
+            default: {
+                if (resp) {
+                    std::string errorMsg = "CTCLoss layer does not support labels precision '"
+                            + std::string(inputs[2]->getTensorDesc().getPrecision().name()) + "'";
+                    errorMsg.copy(resp->msg, sizeof(resp->msg) - 1);
+                }
+                return GENERAL_ERROR;
+            }
+        }
+    }
+
+    template<typename I1, typename I2>
+    StatusCode processData(
+                std::vector<Blob::Ptr>& inputs,
+                std::vector<Blob::Ptr>& outputs,
+                ResponseDesc* resp) noexcept {
+        const float* logits = inputs[0]->cbuffer().as<const float*>() +
+            inputs[0]->getTensorDesc().getBlockingDesc().getOffsetPadding();
+        const I1* logitsLength = inputs[1]->cbuffer().as<const I1*>() +
+            inputs[1]->getTensorDesc().getBlockingDesc().getOffsetPadding();
+        const I2* labels = inputs[2]->cbuffer().as<const I2*>() +
+            inputs[2]->getTensorDesc().getBlockingDesc().getOffsetPadding();
+        const I1* labelsLength = inputs[3]->cbuffer().as<const I1*>() +
+            inputs[3]->getTensorDesc().getBlockingDesc().getOffsetPadding();
+        float* dstData = outputs[0]->buffer().as<float*>() +
+            outputs[0]->getTensorDesc().getBlockingDesc().getOffsetPadding();
+
+        const auto& logitsShape = inputs[0]->getTensorDesc().getDims();
+        const auto batchNum = logitsShape[0];
+        const auto maxTime = logitsShape[1];
+        const auto classesNum = logitsShape[2];
+
+        I2 blankIndex = classesNum - 1;
+        if (inputs.size() > 4) {
+            blankIndex = inputs[4]->cbuffer().as<const I2*>()[0];
+        }
+
+        std::vector<I2> targetD(maxTime);
+        std::vector<I2> pathS(maxTime);
+
+        const size_t TC = maxTime * classesNum;
+
+        for (size_t b = 0; b < batchNum; b++) {
+            I1 actualLogitLen = logitsLength[b];
```

Comment:
May I ask you to use ```const``` here and below if it is possible?

Reply:
Fixed